Fix for Blizzard 'SetMapByID' quirk + Updated Blizzard constant name

MMOSimca opened the pull request:
Added workaround for a Blizzard quirk where SetMapByID can not set the map for several different mapIDs. Also, updated the constant WORLDMAP_WORLD_ID to WORLDMAP_AZEROTH_ID now that Blizzard has moved it.

luis-ortiz commented on Astrolabe.lua:
```diff
@@ -125,7 +125,7 @@ local issecurevariable = issecurevariable
 local function GetCurrentMapAreaID()
 	local id = real_GetCurrentMapAreaID();
-	if ( id < 0 and GetCurrentMapContinent() == WORLDMAP_WORLD_ID ) then
+	if ( id < 0 and GetCurrentMapContinent() == WORLDMAP_AZEROTH_ID ) then
```
Yeah, WORLDMAP_WORLD_ID has gone the way of the dodo.
I wonder why more things have not broken with users of WORLDMAP_WORLD_ID ?

These changes are mandatory.

luis-ortiz commented on Astrolabe.lua:
```diff
@@ -1271,6 +1271,87 @@ local function activate( newInstance, oldInstance )
+		HARDCODED_MAP_INFORMATION_FOR_UNOBTAINABLE_MAPIDS[971][0].BRx = BRx
+		HARDCODED_MAP_INFORMATION_FOR_UNOBTAINABLE_MAPIDS[971][0].BRy = BRy
+		
+		-- Alternate mapIDs for 971 (Alliance Garrison)
```
Under what conditions do the alternate map IDs show up?

luis-ortiz commented:
Interesting.  This patch captures more information about missing maps than my pull request, but mine captures more information about missing floors.   We need both to have TomTom work right.

luis-ortiz commented:
BTW, I have updated my pull request to incorporate the information from @MMOSimca in pull request #1 and put it into my static dumb update style in #2 .

MMOSimca commented:
I believe my latest commit fixes the remaining issues with missing data. Testing that now.

MMOSimca commented:
Looks solid. Note that many of the things you were adding to WorldMapSize are now found under MicroDungeonSize instead and thus, if you want to access them without knowing the terrainMapID ("originSystem"), then you'll need to do something like:

MicroDungeonSize[WorldMapSize[MAPID].originSystem][FLOORINDEX]

I did it this way because WorldMapSize is designed to have floors in order from 0 to numFloors without any outliers, and all of these floors are not considered by the game's API when it counts the number of floors. MicroDungeonSize is designed around floors being out of order or outliers, so it fits in there better.

Also two other things I noticed:
1) Some of the data has different values for xOffset and yOffset than the ones you hardcoded. I believe this is because after the 'fix' to the code, it now puts all of the data through "TRANSFORMS" first. I think this is probably a good thing and may help accuracy rather than harm it but I suppose it remains to be seen.
2) In one instance (possibly more, didn't check every single floor), your data was putting floors on the wrong map. 976 has no floors. Floor 27 is part of map 941 (976's parent map). Just something to keep in mind as a change between my version and yours.

MMOSimca commented:
Refactored the code a bit and added a note about the issue we discussed. As far as I can see, it can never occur currently (all remapped maps have the same originSystem as their working counterpart), but Blizzard could easily add an exception in the future.

I think this solution is now 'complete'... except did you need external access to WorldMapSize/MicroDungeonSize/zeroData or was that just for debugging?

luis-ortiz commented:
That was all for debugging testing if we haved bumped into terra incognita.

I’ll layer a different change to do them same after yours.

But, I better hit my raid for the weekend soon!

> On Mar 22, 2015, at 3:15 PM, MMOSimca notifications@github.com wrote:
> 
> Refactored the code a bit and added a note about the issue we discussed. As far as I can see, it can never occur currently (all remapped maps have the same originSystem as their working counterpart), but Blizzard could easily add an exception in the future.
> 
> I think this solution is now 'complete'... except did you need external access to WorldMapSize/MicroDungeonSize/zeroData or was that just for debugging?
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/jnwhiteh/astrolabe/pull/1#issuecomment-84681387.

luis-ortiz commented:
I will start testing this to see if it subsumes my pull request #1  .

luis-ortiz commented:
FYI:   http://wow-pro.com/node/3626#comment-54514

MMOSimca commented:
I just fixed several issues with my refactored remapping code (I guess I was too tired when I wrote that part, heh). Ran the addon through Globe (http://www.townlong-yak.com/globe/) and it picked up an obscure debug function error (probably has existed for like 5 years, lol) so I fixed that too.

Going to merge this pull request now. Anything further we can fix in the main repo.
